### DISCUSSION TITLE
macOS (Sonoma/Sequoia) build

### DISCUSSION THREAD

**Discussion by shuang886** [ldicker83/Micropolis-SDLPP#4] [General]
Not sure if anybody cares about macOS builds, but here's what I had to do:

- Download `SDL2.framework`, `SDL2_ttf.framework`, and `SDL2_image.framework` from https://github.com/orgs/libsdl-org/repositories and put them in the `xcode` directory. (Or elsewhere if you modify the xcodeproj to point to them.)
- Download Raleway fonts from https://fonts.google.com/specimen/Raleway/about, and copy `Raleway-Medium.ttf` into the `res` directory, but renamed as `raleway-medium.ttf`.
- Change the project's signing team to "None" or your own team.
- Comment out the line:

```C++
    mTexture(loadTexture(renderer, "images/EvalWindow.png")),
```

in `src/EvaluationWindow.cpp` because `EvalWindow.png` appears to be missing.

- Add:

```xml
	<key>com.apple.security.cs.disable-library-validation</key>
	<true/>
```

to `xcode/Micropolis-SDL2PP/Micropolis_SDL2PP.entitlements`, I think because the SDL frameworks were signed by them and the hardened runtime refuses to link it.

- Delete the `AppDelegate` object from `MainMenu.xib` because it doesn't exist.

I put most of these fixes in https://github.com/shuang886/Micropolis-SDL2PP/tree/macos.

**Comment by tuler** [comment]
Thank you

**Comment by ldicker83** [comment]
Been working on it, updating to the latest Xcode but seems appdelegate files are missing? Builds fine but instantly exits because of missing nsobject references or something to that effect. Definitely want to get this working again though!

**Comment by ldicker83** [comment]
Welp, I feel dumb now. I forgot that I added a bunch of additional fonts and that's why it's failing.

Anyway, I do intend to generally keep the Xcode projects up to date, going to be pushing some changes here shortly to get it in line with the Windows builds.

**Comment by shuang886** [comment]
Updating for Xcode 16.4, macOS Sequoia 15.5, commit 93dfdddbc7ff87991e46904fa40b714c25009a5a:

- Download `SDL2.framework`, `SDL2_ttf.framework`, and `SDL2_image.framework` from https://github.com/orgs/libsdl-org/repositories and put them in the `xcode` directory. (Or elsewhere if you modify the xcodeproj to point to them.)
- Download Raleway fonts from https://fonts.google.com/specimen/Raleway/about, and copy `Raleway-Medium.ttf`, `Raleway-Bold.ttf`, and `Raleway-BoldItalic.ttf` into the `res` directory.
- Apply this fix for a couple of missing source files: https://github.com/ldicker83/Micropolis-SDL2PP/pull/16
